### ISSUE TITLE
Mirroring beginning and ending frames

### ISSUE THREAD

**Issue by warnerwarner** [NICALab/SUPPORT#12]
Hi, I understand that by construction the model will remove the first and last n frames from a tif. However, I have trial data within a few frames of the beginning of my recordings and don't want to loose information. I've mirrored the first and last n frames and added them to my data. Is there a foreseeable problem with this?

**Comment by SteveJayH** [comment]
Did you trained with mirrored data, or train with original data and infer with mirrored data?

In the case of training is done in original data, and inference is held in mirrored data,
Denoising of mirrored parts will not accurate. This is because SUPPORT exploits temporal information, the order of frames is important in the processing. If we think about calcium imaging, the temporal pattern of linear increase and exponential decrease will flipped, so the data characteristic of mirrored parts are different to the original part. So the mirrored part is the **unseen** to SUPPORT, which may lead degraded performance. 

If we use mirrored data in the training, mirrored parts are now seen to the network. However, this mirrored parts are still differ to the original area. In the ideal case, it is possible that the network learns good denoising strategy even the frame orders are different. Or, there is a possibility that the network confuse of the frame orders and do not use temporal patterns, which may hinder denoising even the original part.

In short, I'm not sure about mirroring and I think that it **may** be possible if we train with mirrored data. However, note that this is not conclusion with the experiments.

FYI, we sometimes use input_frames as 11, instead of 61, in the case of processing data with small number of frames.

Please let me know if you have any thoughts, or any changes!
